ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml

```diff
@@ -38,31 +40,34 @@ Item {
         }
     }
 
-
     function isAcceptedItem(item) {
         let properties = this.property.split(",")
             .map(p => p.trim())
             .filter(p => !!item[p])
 
-        if (properties.length == 0) {
+        if (properties.length == 0 || this.filter.length == 0 || this.cursorPosition == 0) {
             return false
         }
 
-        if (this.filter.endsWith("@")) {
-            return true
+        // Prevents suggestions to show up at all
+        if (this.filter.indexOf("@") === -1)  {
+          return false
         }
 
-        let lastAt = this.filter.lastIndexOf("@")
-
-        if (lastAt == -1) {
-            return false
+        let cursorAtEnd = this.cursorPosition === this.filter.length;
+        let hasAtBeforeCursor = this.filter.charAt(this.cursorPosition - 1) === "@" 
+        let hasWhiteSpaceBeforeAt = this.filter.charAt(this.cursorPosition - 2) === " "
+        let hasWhiteSpaceAfterAt = this.filter.charAt(this.cursorPosition) === " "
+
+        if (this.filter == "@" ||
+          (hasAtBeforeCursor && hasWhiteSpaceBeforeAt && hasWhiteSpaceAfterAt) ||
+          (this.cursorPosition === 1 && hasAtBeforeCursor && hasWhiteSpaceAfterAt) ||
+          (cursorAtEnd && this.filter.endsWith("@") && hasWhiteSpaceBeforeAt)) {
+          this.lastAtPosition = this.cursorPosition - 1;
+          return true
```

Comment:
Interesting that we are allowing someone to type `@ ` (an `@` plus a space) and still have the suggestions come up. I would think that `@ ` would mean the user didn't want to type a suggestion and just wanted to type an `@` in the text. IOW, if I type `@`, a suggestion box appears, then I type `space`, the suggestion popup should disappear as if I didn't want to use the suggestion box. Maybe I'm reading this incorrectly?

Reply:
> (an @ plus a space) and still have the suggestions come up

Actually, this particular condition ensures that the auto complete only opens up if:
1. There's a space after the `@` in any case (e.g. we don't want it to open at `Hello @|you`
2. There's a space before the `@` (e.g. we don't want it to open at `Hello you@|`
3. If the `@` is at the beginning of the message, there can't be a space (`@|`)

> IOW, if I type @, a suggestion box appears, then I type space, the suggestion popup should disappear as if I didn't want to use the suggestion box. Maybe I'm reading this incorrectly?

That's what I would expect as well and you're right, at the moment it doesn't behave that way. 
Will update!

Reply:
Done and rebased

Reply:
For #1, if the cursor is between `@` and a word, ie `@|you`, do you think then it might be a good idea for the suggestion box to open with `you` as the suggestion box filter?

Reply:
Personally, don't have too strong opinions about it. I checked how other messengers are handling this and implemented that. If you feel the suggestion box should open up either way, I'm happy to change that :)